generators/app/templates/src/app/login/__auth._login.component.ts

```diff
@@ -41,40 +42,38 @@ export class LoginComponent implements OnInit {
 
   ngOnInit() { }
 
+  ngOnDestroy() { }
+
   login() {
+    this.isLoading = true;
+    const login$ = this.authenticationService.login(this.loginForm.value);
 <% if (props.ui === 'ionic') { -%>
-<% /**
-     * currently a workaround for ionic zone handling - needs rewrite for readability.
-     * See: https://github.com/ngx-rocket/generator-ngx-rocket/pull/369#discussion_r217625108
-     * Revert https://github.com/ngx-rocket/generator-ngx-rocket/pull/369/commits/4969a42a7a56a03e699498c70d987a23eea1aee4
-     * when ionic overlay zone handling works again
-     * Also see https://github.com/angular/zone.js/issues/1142
-     */
--%>
-    // const loadingPromise = this.loadingController.create();
-    // const loadingPresentedPromise = loadingPromise
-    //  .then(loading => loading.present());
+    const loading$ = from(
+      this.loadingController.create().then(loadingOverlay => {
+        this.loadingOverlay = loadingOverlay;
+        return loadingOverlay.present();
+      })
+    );
+    forkJoin(login$, loading$).pipe(
+      map(([credentials, ...rest]) => credentials),
 <% } else { -%>
-    this.isLoading = true;
+    login$.pipe(
 <% } -%>
-    this.authenticationService.login(this.loginForm.value)
-      .pipe(finalize(() => {
+      untilDestroyed(this),
```

Comment:
untilDestroyed() should always be the last operator in the chain to avoid possible leaks with intermediate observables